commit 1972460bf7ad803097f8fb669dcb858f6135c7d5
Author: Karel Klic <kklic@redhat.com>
Date:   Mon Mar 28 18:19:01 2011 +0200

    abrt-action-analyze-backtrace: report failure when run on a dump_dir without backtrace

diff --git a/src/plugins/abrt-action-analyze-backtrace.c b/src/plugins/abrt-action-analyze-backtrace.c
index b690fc9..7ae7e47 100644
--- a/src/plugins/abrt-action-analyze-backtrace.c
+++ b/src/plugins/abrt-action-analyze-backtrace.c
@@ -88,7 +88,13 @@ int main(int argc, char **argv)
     char *executable = dd_load_text(dd, FILENAME_EXECUTABLE);
 
     /* Read backtrace */
-    char *backtrace_str = dd_load_text(dd, FILENAME_BACKTRACE);
+    char *backtrace_str = dd_load_text_ext(dd, FILENAME_BACKTRACE,
+                                           DD_LOAD_TEXT_RETURN_NULL_ON_FAILURE);
+    if (!backtrace_str)
+    {
+        dd_close(dd);
+        return 1;
+    }
 
     /* Compute backtrace hash */
     struct btp_location location;
